### PR TITLE
Fix Navigation bug with dimensions of Matrices

### LINKED ISSUE #68: Kalman Multivariate Dimensions
Right Now the code in Kalman Multivariate allows changing the internal matrices needed to perform calculations freely. This might result in changing a matrix of dimension n x n to a matrix m x k and will result in a rather complicated exception from the Eigen library. The code should take care of this by raising its own exception earlier and not in the multiplication.

### PR DESCRIPTION
**Pull request by pmorande27**
## Description
The code for KalmanMultivariate now checks if the dimension of the given matrices correspond to the expected dimensions, if there is a matrix (or vector) with wrong dimensions is passed to the programme it will throw an exception. It is up to Navigation Team to decide what to do with that exception. However, this situation is not expected unless someone breaks the code by passing directly a wrong matrix to the programme. Therefore, it should not affect the current version in develop.

## Changes
- KalmanMultivariate.cpp: Now in all the methods that reassign matrices or vectors there is a checker that will look if the dimensions of the given objects are correct.
## Additional Information
Fixes #68

### CODE REVIEW

File: src/utils/math/kalman_multivariate.cpp [hunk 1]
```diff
@@ -39,21 +39,44 @@ namespace hyped {
 
             void KalmanMultivariate::setDynamicsModel(MatrixXf& A, MatrixXf& Q)
             {
+                if (A.cols()!= n_ || A.rows()!= n_ || Q.cols()!=n_ ||  Q.rows()!=n_ )
```
Comment (DonaldJennings) [inline src/utils/math/kalman_multivariate.cpp]: Try reformatting these conditional statements so they're more aligned with the style guide. Mainly the curly braces should be on the same line. [More info here](https://hyp-ed.github.io/styleguide/#Conditionals)

**Review by DonaldJennings — CHANGES_REQUESTED** [review]
Overall functionality looks great, just some formatting issues to change. What would be quite nice before merging is if we made some tests so that we check whether the exception was thrown for invalid parameters. You would need to check with Cal and Flori on their plans for Kalman testing to ensure there won't be any conflicts but that shouldn't take too long to implement :)

File: src/utils/math/kalman_multivariate.cpp (hunk 1 above)
Comment (pmorande27) [reply src/utils/math/kalman_multivariate.cpp]: I think it should be more or less fixed now, linter is not working when I am trying to push on this branch which is weird

Comment (pmorande27) [reply src/utils/math/kalman_multivariate.cpp]: Also, I already have a version of tests for these exceptions, I am waiting for these changes to be on develop to paste them in the new branch of Kalman tests, right now I have two new ones

File: src/utils/math/kalman_multivariate.cpp [hunk 2]
```diff
@@ -72,19 +85,31 @@ namespace hyped {
 
             void KalmanMultivariate::updateA(MatrixXf& A)
             {
+                if (A.cols() != n_ || A.rows() != n_) {
+                  throw  std::invalid_argument("Wrong dimension of the Matrices");
+                } else {
                 A_ = A;
```
Comment (DonaldJennings) [inline src/utils/math/kalman_multivariate.cpp]: Code within curly braces should be two spaces in from the edge :)

**Review by DonaldJennings — CHANGES_REQUESTED** [review]
Just one other minor formatting change needed for consistency

File: src/utils/math/kalman_multivariate.cpp (hunk 1 above)
Comment (DonaldJennings) [reply src/utils/math/kalman_multivariate.cpp]: Is the linter not working at all? Does it come up with anything linter-related or just doesn't run?

File: src/utils/math/kalman_multivariate.cpp (hunk 2 above)
Comment (pmorande27) [reply src/utils/math/kalman_multivariate.cpp]: so I should add another space right?

File: src/utils/math/kalman_multivariate.cpp (hunk 1 above)
Comment (pmorande27) [reply src/utils/math/kalman_multivariate.cpp]: It runs if I make --make lint but it is not running on the commit

File: src/utils/math/kalman_multivariate.cpp (hunk 2 above)
Comment (DonaldJennings) [reply src/utils/math/kalman_multivariate.cpp]: for the A_ line yea that should be 2 spaces in from the left curly brace like follows
image

Comment (pmorande27) [reply src/utils/math/kalman_multivariate.cpp]: done I think

File: hyped.code-workspace [hunk 3]
```diff
@@ -35,5 +35,8 @@
     "editor.detectIndentation": false,
     "C_Cpp.default.cppStandard": "c++11",
     "C_Cpp.default.cStandard": "c11",
+    "files.associations": {
```
Comment (DonaldJennings) [inline hyped.code-workspace]: This shouldn't be here : ) Other than that looking good

**Review by DonaldJennings — APPROVED** [review]

File: hyped.code-workspace (hunk 3 above)
Comment (pmorande27) [reply hyped.code-workspace]: Done now :)

**Review by alv31415 — COMMENTED** [review]
All changes look good.

**Review by alv31415 — APPROVED** [review]

**Pull request merged**